Popularity/Kipras/pyspark/pyspark_ml.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+
+# File       : pyspark_ml.py
+# Author     : Kipras Kancys <kipras [DOT] kan  AT gmail [dot] com>
+# Description: pysmark model
+
+# system modules
+import time
+import argparse
+
+# pyspark modules
+from pyspark import SparkContext
+from pyspark.sql import SQLContext, Row
+from pyspark.sql.types import StructType, StructField, LongType
+
+from pyspark.ml import Pipeline
+from pyspark.ml.feature import StringIndexer, VectorIndexer, Binarizer
+from pyspark.ml.classification import RandomForestClassifier
+
+from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+
+from pyspark.mllib.regression import LabeledPoint
+
+class OptionParser():
+    def __init__(self):
+        "User based option parser"
+        self.parser = argparse.ArgumentParser(prog='PROG', description=__doc__)
+        self.parser.add_argument("--clf", action="store",
+            dest="clf", default="", help="Classifier to use")
+        self.parser.add_argument("--train", action="store",
+            dest="ftrain", default="", help="Input train file")
+        self.parser.add_argument("--valid", action="store",
+            dest="fvalid", default="", help="Input validation file")
+        self.parser.add_argument("--prediction", action="store",
+            dest="fprediction", default="output.txt", help="Output file for predictions")
+
+class SparkLogger(object):
+    "Control Spark Logger"
+    def __init__(self, ctx):
+        self.logger = ctx._jvm.org.apache.log4j
+        self.rlogger = self.logger.LogManager.getRootLogger()
+
+    def set_level(self, level):
+        "Set Spark Logger level"
+        self.rlogger.setLevel(getattr(self.logger.Level, level))
+
+    def lprint(self, stream, msg):
+        "Print message via Spark Logger to given stream"
+        getattr(self.rlogger, stream)(msg)
+
+    def info(self, msg):
+        "Print message via Spark Logger to info stream"
+        self.lprint('info', msg)
+
+    def error(self, msg):
+        "Print message via Spark Logger to error stream"
+        self.lprint('error', msg)
+
+    def warning(self, msg):
+        "Print message via Spark Logger to warning stream"
+        self.lprint('warning', msg)
+
+def prepData(sqlContext, ctx, fname):
+    "Load, add label col and convert data into label and feature type dataframe (needed for MLlib)"
+
+    lines = ctx.textFile(fname)
+    parts = lines.map(lambda l: l.split(","))
+
+    # to set col names:
+    newColumns = parts.first()
+    parts = parts.filter(lambda x:x !=newColumns)
+    df = parts.toDF()
+    oldColumns = df.schema.names
+    df = reduce(lambda df, idx: df.withColumnRenamed(oldColumns[idx], newColumns[idx]), xrange(len(oldColumns)), df)
+
+    # make labeled points, needed for algorithms to work
+    temp = df.map(lambda row: LabeledPoint(row[-1], row[:-1]))
```

Comment:
I checked that. labeledPoint is made from NumPy array. Which means all values inside are treated as numerical values.